### DISCUSSION TITLE
Add ability to run setup without Cargo.toml

### DISCUSSION THREAD

**Discussion by yunmikun2** [diesel-rs/diesel#3217] [Ideas]
The tool is pretty suitable to be used as a stand alone migration management. It almost doesn't have any dependency on any framework or tool. It's just a binary that runs SQL scripts.

Now it uses `Cargo.toml` as a marker for a project root. So when you want to use it as a stand alone migration tool, you need to create an empty `Cargo.toml` file run `diesel setup` and then delete the file.

It would be nice to either have a flag that says that the current directory is indeed a project root, or being able to detect other types of projects (like, for example, a git repo).

**Comment by yunmikun2** [comment]
Judging by this line https://github.com/diesel-rs/diesel/blob/bc8d85d3621a6492a9a9786716a680809584f84c/diesel_cli/src/main.rs , it seems like creating an empty `diesel.toml` file should be sufficient too. But it still be better if a user doesn't need to create any files prior to executing a setup action.

**Comment by weiznich** [comment]
I would be open to merge a well written PR with tests and documentation for this. Otherwise this is currently not a priority for me.
